scheds/kernel-examples/scx_central.bpf.c

```diff
@@ -142,6 +142,13 @@ static bool dispatch_to_cpu(s32 cpu)
 	s32 pid;
 
 	bpf_repeat(BPF_MAX_LOOPS) {
+		/* We might run out of dispatch buffer slots if we continue dispatching
+		 * to the fallback DSQ, without dispatching to the local DSQ of the
+		 * target CPU. In such a case, break the loop.
+		 */
+		if (!scx_bpf_dispatch_nr_slots())
```

Comment:
Given that we know that there are slots on entry to the function, would it be better to check nr_slots in the FALLBACK_DSQ block right before continue?

Reply:
Hmm, but I was thinking of a case where say you have 2 slots remaining, you pull out 2 tasks from the central_q which go to the fallback DSQ, and now you pull a 3rd which can go to the target CPU's local DSQ, but it will error out. So while it is true we have slots on entry it may not be true for the SCX_DSQ_LOCAL_ON case for later iterations.

Does that make sense?

Reply:
Hmm... so, we have either:
```
while (true) {
  if (no slot)
    break;
  find task;
  if (cpu doesn't match) {
    dispatch(FALLBACK);
    continue;
  }
  dispatch(target_cpu);
  break;
}
```

or 

```
while (true) {
  find task;
  if (cpu doesn't match) {
    dispatch(FALLBACK);
    if (more slots)
      continue;
    else
      break;
  }
  dispatch(target_cpu);
  break;
}
```

If we know that there are slots on entry, the behavior should be identical between the two, right? The only difference is that the former would do an extra check which will always indicate more slots.